internal/provider/common.go

```diff
@@ -62,6 +59,109 @@ func isUMNMandatoryForCluster(cluster *common.Cluster) bool {
 	return cluster != nil && isUMNMandatoryForPlatform(cluster.Platform)
 }
 
+// set default values for the external platform
+func setExternalDefaultValues(platform *models.Platform, cluster *common.Cluster) {
+	if platform == nil {
+		// nothing to update
+		return
+	}
+
+	if platform.Type != nil && *platform.Type != models.PlatformTypeExternal {
+		// not an external platform
+		return
+	}
+
+	if platform.Type == nil && cluster != nil && *cluster.Platform.Type != models.PlatformTypeExternal {
+		// not an external platform
+		return
+	}
+
+	// set platform type to force the update of the platform later in DB
+	platform.Type = common.PlatformTypePtr(models.PlatformTypeExternal)
+
+	// we are creating a new cluster, set CloudControllerManager to "" if unset
+	if cluster == nil && platform.External.CloudControllerManager == nil {
+		platform.External.CloudControllerManager = swag.String(models.PlatformExternalCloudControllerManagerEmpty)
+	}
+
+	// we are updating an existing cluster from a non-external cluster
+	// we must ensure that CloudControllerManager is set
+	if cluster != nil && *cluster.Platform.Type != models.PlatformTypeExternal && platform.External.CloudControllerManager == nil {
+		platform.External.CloudControllerManager = swag.String(models.PlatformExternalCloudControllerManagerEmpty)
+	}
+}
+
+func areExternalSettingsSet(platform models.Platform) bool {
+	return platform.External != nil && (platform.External.PlatformName != nil || platform.External.CloudControllerManager != nil)
+}
+
+func validateExternalPlatformName(platformName *string) bool {
+	return platformName != nil && len(*platformName) > 0
+}
+
+func validateExternalCloudControllerManager(cloudControllerManager string) bool {
```

Comment:
This is already defined in the swagger, do we need this here too?

Reply:
true, for some reasons it was not working during early testings, that's why I added the check. But I saw the swagger validation work yesterday.